compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -271,35 +272,37 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
         Architecture::Riscv32 | Architecture::Riscv64 => {
             // Source: https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/079772828bd10933d34121117a222b4cc0ee2200/riscv-elf.adoc
             let mut e_flags: u32 = 0x0;
-            let features = &sess.target.options.features;
+
             // Check if compressed is enabled
-            if features.contains("+c") {
+            if sess.unstable_target_features.contains(&sym::c) {
```

Comment:
I'm not sure whether looking at the `unstable_target_features` is correct here.

Actually, I don't understand that field at all. Previously, the only usage of it seems to be when deciding on the feature available inside an inline asm context. But I also see other inline asm related code using `sess.target_features` normally. Unstable features also seem especially unfortunate because they're silently discarded from `sess.target_features` on stable compilers, but that's besides the point.

The changes below look good to me though.

Reply:
But wouldn't `target_features` mean that "+c" is silently ignored in non-nightly compilers?  

TBH the whole `target_features` unstable computation thing is puzzling to me.

Reply:
> But wouldn't target_features mean that "+c" is silently ignored in non-nightly compilers?

well, it is unstable according to the list so that's "correct behavior" (but pretty stupid behavior)

> TBH the whole target_features unstable computation thing is puzzling to me.

same.

so given that `+c` used to work on stable already I'm fine with keeping `unstable_target_features` here. Can you at least add a comment saying that using `unstable_target_features` was chosen because it used to work on stable?

pinging @Amanieu because maybe you have an idea: What exactly is the deal with unstable target features? They're silently ignored on stable which is pretty odd and also in one place they're actually used on stable.

Reply:
The "unstable features" still gets passed to LLVM (well, actually, it's the other way around, they are worked out from LLVM in the first place).